lib/shoulda/matchers/action_controller/callback_matcher.rb

```diff
@@ -188,8 +321,38 @@ def callbacks
           end
         end
 
+        def filter_matches?
+          callbacks.map(&:filter).include?(method_name)
+        end
+
+        def only_matches?
+          if options.key?(:only)
+            callbacks.any? do |c|
+              c.filter == method_name && callback_if(c) == only_options_as_if
+            end
+          else
+            true
+          end
+        end
+
+        def callback_if(callback)
+          callback.instance_variable_get(:@if).first
```

Comment:
It is only a array with a single item, with item is something like `"action_name == 'index' || "action_name == 'edit'"`.
I will double check this, I need to see why this is broken on rails 4.0.0 and 4.0.1.